src/OpenConext/ProfileBundle/Service/SpecifiedConsentListService.php

```diff
@@ -21,50 +23,35 @@
 use OpenConext\EngineBlockApiClientBundle\Service\AttributeReleasePolicyService;
 use OpenConext\Profile\Entity\AuthenticatedUser;
 use OpenConext\Profile\Value\SpecifiedConsentList;
+use Symfony\Component\Security\Core\User\UserInterface;
 
 class SpecifiedConsentListService
 {
-    /**
-     * @var ConsentService
-     */
-    private $consentService;
-
-    /**
-     * @var AttributeReleasePolicyService
-     */
-    private $attributeReleasePolicyService;
-
-    /**
-     * @param ConsentService $consentService
-     * @param AttributeReleasePolicyService $attributeReleasePolicyService
-     */
     public function __construct(
-        ConsentService $consentService,
-        AttributeReleasePolicyService $attributeReleasePolicyService
+        private readonly ConsentService $consentService,
+        private readonly AttributeReleasePolicyService $attributeReleasePolicyService,
     ) {
-        $this->consentService                = $consentService;
-        $this->attributeReleasePolicyService = $attributeReleasePolicyService;
     }
 
-    public function getListFor(AuthenticatedUser $user): SpecifiedConsentList
-    {
-        $consentList = $this->consentService->findAllFor($user);
+    public function getListFor(
+        UserInterface $user,
+    ): SpecifiedConsentList {
+        assert($user instanceof AuthenticatedUser);
 
-        // There is an off chance the user didn't give consent yet, in that case the consent list is null. It's not
-        // possible to apply ARP on empty consent list.
-        if (is_null($consentList)) {
-            // So return an empty SpecifiedConsentList
-            return SpecifiedConsentList::createWith([]);
-        }
+        $consentList = $this->consentService->findAllFor($user);
 
         return $this->attributeReleasePolicyService->applyAttributeReleasePolicies(
             $consentList,
-            $user->getAttributesFiltered()
+            $user->getAttributesFiltered(),
         );
```

Comment:
Logic actually changed here. The API can renturn an empty set but the code no longer handles that situation right?

Reply:
No, this always was false, as findAllFor always returns a ConsentList. Was reported by phpstan I think

Reply:
Ok, if you did your investigative work. I'm fine with the change!